src/platform/ESP32/ConfigurationManagerImpl.cpp

```diff
@@ -161,18 +173,34 @@ CHIP_ERROR ConfigurationManagerImpl::Init()
         SuccessOrExit(err);
     }
 
-    if (!ESP32Config::ConfigValueExists(ESP32Config::kCounterKey_TotalOperationalHours))
+    if (CHIP_NO_ERROR != GetTotalOperationalHours(mTotalOperationalHours))
     {
-        err = StoreTotalOperationalHours(0);
+        err = StoreTotalOperationalHours(mTotalOperationalHours);
         SuccessOrExit(err);
     }
 
+    {
+        // Start a timer which reloads every one hour and bumps the total operational hours
+        TickType_t reloadPeriod   = (1000 * 60 * 60) / portTICK_PERIOD_MS;
```

Comment:
Is it required to add a timer specifically for this? Is there any IDF API that gives time (ms) since bootup? If yes, can we just use that and convert to hrs when a request is made?

Reply:
Yes, it is required to add the timer for this. I think you are talking about uptime which is time since bootup. This is total operational hours, and needs to be bumped every hour, and must be persisted across reboots. There is no API for that in esp-idf.